lib/abstract_format.ml

```diff
@@ -53,16 +53,26 @@ and pattern_assoc_t =
 and expr_t =
   | ExprBody of expr_t list
   | ExprCase of line_t * expr_t * clause_t list
+  | ExprLocalFunRef of line_t * string * int
+  | ExprRemoteFunRef of line_t * atom_or_var_t * atom_or_var_t * integer_or_var_t
+  | ExprFun of line_t * string option * clause_t list
   | ExprLocalCall of line_t * expr_t * expr_t list
   | ExprRemoteCall of line_t * line_t * expr_t * expr_t * expr_t list
   | ExprMapCreation of line_t * expr_assoc_t list
   | ExprMapUpdate of line_t * expr_t * expr_assoc_t list
+  | ExprMatch of line_t * pattern_t * expr_t
   | ExprBinOp of line_t * string * expr_t * expr_t
   | ExprVar of line_t * string
   | ExprLit of literal_t
 and expr_assoc_t =
   | ExprAssoc of line_t * expr_t * expr_t
   | ExprAssocExact of line_t * expr_t * expr_t
+and atom_or_var_t =
+  | Atom of line_t * string
```

Comment:
I think the name `Atom` is too general to express "atom variant of atom or var".
How about `AtomVarAtom` and `AtomVarVar` ?

Reply:
I renamed `Atom` to `AtomVarAtom`, `AtomVar` to `AtomVarVar`, and so on.